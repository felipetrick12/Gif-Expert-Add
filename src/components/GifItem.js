import React from 'react'
import PropTypes from "prop-types";


export default function GifItem({title,url}) {

    return (
        
        <div className='card animate__animated animate__zoomInDown' >
           <p>{title}</p>
           <img src={url} alt={title}/>
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
  };
