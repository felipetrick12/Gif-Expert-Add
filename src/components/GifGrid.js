import React from "react";
import { useEffectGitch } from "../hooks/useEffectGitch";
import GifItem from "./GifItem";
import PropTypes from 'prop-types';


export const GifGrid = ({ category }) => {
   
       const {data:imagen,loading}= useEffectGitch(category);


        return (
            <>

            
            <h2>{category}</h2>
            <p>{loading && 'Cargando'}</p>
            
                <div className='cardGrid '>
                {imagen.map((img) => (
                    <GifItem key={img.id} {...img}/> //se utiliza el expred para enviar propiedades indepenedientes del arreglo
                ))}
               </div>
           
            </>
        );
};
GifGrid.prototype = {
    category: PropTypes.string.isRequired
}
