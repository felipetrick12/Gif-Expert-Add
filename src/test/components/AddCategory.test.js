import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
import React from 'react';
import { AddCategory } from '../../components/AddCategory';



describe('Debe agregar una nueva categoria', () => {
    const funcion =jest.fn();
    let wrapper = shallow(<AddCategory setCategories={funcion}/>);

    beforeEach(()=> { //ciclo de vida, para que un objeto se reinicialize
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={funcion}/>);
    })

    test('debe pasar el componente ', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe cambiar el valor del componente ', () => {
        
        const value= '';
        const input= wrapper.find('input');
        input.simulate('change', {target: value}); //en el inpu simula que el input es cambiado, que dentro de ese change esta buscando un evento target y el target trae ese valor
        expect(wrapper.find('p').text().trim()).toBe(value);
    })
    
    test('no debe de enviar la informacion si es menor que 2 ', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){}})

        expect(funcion).not.toHaveBeenCalled();
    });

    test('debe de llamar la funcion  y limpiar la caja de texto ', () => {
        const value= 'hola'; 
        
        wrapper.find('input').simulate('change', {target: {value}}); //simula el input,y sus propiedades
        
        wrapper.find('form').simulate('submit', {preventDefault(){}}) //simula el form y sus propiedades,
        expect(funcion).toHaveBeenCalled(); //espera que la funcion se halla llamado por lo menos una vez

        expect(wrapper.find('input').prop('value')).toBe('') //mira si el valor esperado es en blanco q es el q limpia la caja
    })
    
    
})
