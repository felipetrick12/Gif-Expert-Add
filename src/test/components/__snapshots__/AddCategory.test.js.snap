// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debe agregar una nueva categoria debe pasar el componente  1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
